### PR TITLE
Simplify A/B test lookup via `express-http-context`

### PR DESCRIPTION
**Pull request by damassi**
See https://github.com/artsy/force/pull/5099 for more info. 

This updates our `getENV` isomorphic ENV lookup util with another server-side layer:
- First, check to see if `SOME_ENV` var exists within the request / response cycle, stored within express-http-context. If so, return that value. Useful for A/B testing.
- If undefined, fall back to `process.env.SOME_ENV`

### CODE REVIEW

File: src/Utils/getENV.tsx [hunk 1]
```diff
@@ -2,6 +2,14 @@ import { data as sd } from "sharify"
 
 export function getENV(ENV_VAR) {
   const isServer = typeof window === "undefined"
-  const envVar = isServer ? process.env[ENV_VAR] : sd[ENV_VAR]
+
+  let envVar
+  if (isServer) {
+    const httpContext = require("express-http-context")
+    envVar = httpContext.get(ENV_VAR) ?? process.env[ENV_VAR]
```
Comment (damassi) [inline src/Utils/getENV.tsx]: Uses new [nullish coalesing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) plugin, which always returns the left hand side of the operation unless `null` or `undefined`.

File: .babelrc [hunk 2]
```diff
@@ -19,6 +19,8 @@
       }
     ],
     "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-proposal-optional-chaining",
```
Comment (damassi) [inline .babelrc]: Suprised this wasn't already in! We can no do 
```
const deepPropOrUndefined = foo?.bar?.baz?.bam
```
which should eliminate our need for the `get` util.

File: src/Apps/Artwork/ArtworkApp.tsx [hunk 3]
```diff
@@ -222,7 +222,14 @@ export class ArtworkApp extends React.Component<Props> {
...
           </Row>
 
-          <div id="lightbox-container" />
+          <div
+            id="lightbox-container"
+            style={{
+              position: "absolute",
+              top: 0,
+              zIndex: 1100, // over top nav
+            }}
```
Comment (damassi) [inline src/Apps/Artwork/ArtworkApp.tsx]: Fixes long standing issue where when a user views an artwork in full screen, the top nav bar flops out due to z-index and position.

File: .babelrc (hunk 2 above)
Comment (mzikherman) [reply .babelrc]: oo sick!

File: src/Apps/Artwork/Components/OtherWorks/Header.tsx [hunk 4]
```diff
@@ -16,14 +17,11 @@ export const Header: React.SFC<HeaderProps> = props => {
...
       {buttonHref && (
-        <Button
-          variant="secondaryOutline"
-          mb={3}
-          // FIXME: Move to <Link> component once https://github.com/artsy/palette/pull/130 is merged
-          onClick={() => (window.location.href = buttonHref)}
-        >
-          View all
-        </Button>
+        <RouterLink to={buttonHref}>
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/OtherWorks/Header.tsx]: Ah cool, yea these custom `window.location.href` navigations in JS dont work with our router.

File: src/Utils/getENV.tsx (hunk 1 above)
Comment (mzikherman) [reply src/Utils/getENV.tsx]: Interesting. I didn't fully get immediately how different is than `||`. I do now recognize the subtle difference, but I think generally any sort of conditional assignment like this, if you are expecting a false-y/null/undefined to be possible (or even valid, esp. as a default), I almost always open a console to try it out (and/or mentally unpack it in my brain for a second).

Cool! I look forward to trying this.

**Comment by mzikherman** [comment]
This looks good! Since this depends on Node v12, which still hasn't been fully deployed to production, so we hold off on merging these PR's until we're happy with v12 in prod?

**Comment by artsyit** [comment]
:rocket: PR was released in `v25.18.1` :rocket:

**Pull request merged**